### PR TITLE
Class based prop tests almost done

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: property_tests/tests/class_syntax/stable_b_tree_map/test/utils.ts
Comment (lastmjs) [inline property_tests/tests/class_syntax/stable_b_tree_map/test/utils.ts:1]: Is this not completely broken now for class-based syntax?

File: property_tests/arbitraries/candid/candid_values_arb.ts [hunk 1]
```diff
@@ -158,7 +159,16 @@ export function CandidValueArb(
...
     if (candidType === 'Service') {
-        return ServiceValueArb(candidTypeMeta as ServiceCandidDefinition);
+        // TODO This seems problematic. I was trying to avoid having to pass the syntax variable through all of the values just so that Service and Opt can work.
+        // TODO But now that we have this here and it looks like we might not be able to avoid it
+        // TODO But we might be able to keep going, I haven't run into any problems yet
+        // TODO but also that's a little weird that I haven't run into any problems yet
+        // TODO but maybe we should have a different way of handling the syntax selection. Like if it's a global setting should we have some global variable instead?
+        // TODO or maybe an environment variable?
+        // TODO all of this also applies to the FunctionalOptValuesArb above
+        return FunctionalServiceValueArb(
```
Comment (lastmjs) [inline property_tests/arbitraries/candid/candid_values_arb.ts:169]: Discuss adding syntax everywhere

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: property_tests/arbitraries/candid/candid_values_arb.ts (hunk 1 above)
Comment (bdemann) [reply property_tests/arbitraries/candid/candid_values_arb.ts:169]: done

File: property_tests/tests/class_syntax/stable_b_tree_map/test/utils.ts
Comment (bdemann) [reply property_tests/tests/class_syntax/stable_b_tree_map/test/utils.ts:1]: done

**Pull request closed without merging**